shears/mine_for_gold.py

```diff
@@ -36,23 +42,57 @@ def main():
     args = parser.parse_args()
 
     print('Authenticating Reddit API connection...')
+    
+    """Authorized Reddit Instances"""
     reddit = praw.Reddit(
         client_id=args.client_id,
         client_secret=args.client_secret,
         username=_USERNAME,
         password=args.password,
         user_agent=_USER_AGENT,
     )
+
+    """ To verify that you are authenticated as the correct user """
+    print(reddit.user.me())     
+    
     print('Authenticated!')
 
+
     print('Mining for gold...')
     whitelist = get_ticker_set()
-    for comment in reddit.subreddit('wallstreetbets').comments():
-        print(comment.author.name + ' says:\n')
-        print(comment.body)
-        print('Ticker counts: ' + repr(scrape_tickers(comment.body,
-                                                      whitelist=whitelist)))
 
+    subredditName = "wallstreetbets"
+
+    updated_json = {}
+    updated_json[subredditName] = []
+
+    try:
+        with open('comments.json') as current_comments: 
+            try:   
+                current_data = json.load(current_comments)
+            except : 
+                current_data = {}
+            new_and_old_comments = []
+            if subredditName in current_data:
+                for dicObj in current_data[subredditName]:
+                    new_and_old_comments.append(dicObj)
+            for comment in reddit.subreddit(subredditName).comments():
+                if comment.id in new_and_old_comments:
+                    continue
+                new_and_old_comments.append({
+                    comment.id :{
+                        "Date": datetime.utcfromtimestamp(comment.created_utc).strftime('%Y-%m-%d %H:%M:%S'),
+                        "Name": comment.author.name, 
+                        "Body": comment.body, 
+                        "Votes": comment.score
+                        } 
+                    })
+            current_data[subredditName] = new_and_old_comments
+            updated_json[subredditName] = current_data[subredditName]
+            with open('comments.json', 'w') as outfile:
```

Comment:
This is a bit confusing as it's nested under acquisition of the same file handle.

I would move this out of the body of the previous with statement, e.g.

open, read, and close the file

then

open, overwrite, and close the file